File: QTD2.Domain/Entities/Core/Procedure_EnablingObjective_Link.cs
namespace QTD2.Domain.Entities.Core
{
    public class Procedure_EnablingObjective_Link : Common.Entity
    {
        public int ProcedureId { get; set; }

        public int EnablingObjectiveId { get; set; }

        public virtual Procedure Procedure { get; set; }

        public virtual EnablingObjective EnablingObjective { get; set; }

        public Procedure_EnablingObjective_Link(Procedure procedure, EnablingObjective enablingObjective)
        {
            ProcedureId = procedure.Id;
            EnablingObjectiveId = enablingObjective.Id;
            Procedure = procedure;
            EnablingObjective = enablingObjective;
        }

        public Procedure_EnablingObjective_Link()
        {
        }
    }
}

File: QTD.SPA/src/app/_DtoModels/Procedure_EnablingObjective_Link/Procedure_EnablingObjective_Link.ts
import { EnablingObjective } from '../EnablingObjective/EnablingObjective';
import { Procedure } from '../Procedure/Procedure';

export class Procedure_EnablingObjective_Link {
  procedureId!: any;
  EOIds!: any[];
  procedure!: Procedure;
  enablingObjective!: EnablingObjective;
}
